### PR TITLE
test(ct): update child

### PR DESCRIPTION
**Pull request by sand4rt**
partial fix for: https://github.com/microsoft/playwright/issues/22328

Updating child props, slots or events in vue2 and vue3 does not work yet. Added the tests for the frameworks where it already worked and i'm planning to reuse those tests for vue2 and vue3.

### CODE REVIEW

File: tests/components/ct-react-vite/src/components/Counter.tsx [hunk 1]
```diff
@@ -17,9 +17,9 @@ import { useLayoutEffect, useRef, useState } from "react"
        _remountCount++;
      }
    }, [])
-   return <div onClick={() => props.onClick?.('hello')}>
+   return <button onClick={() => props.onClick?.('hello')}>
```
Comment (pavelfeldman) [inline tests/components/ct-react-vite/src/components/Counter.tsx:20]: button with divs inside does not sound right - what are you trying to do here?

Comment (sand4rt) [reply tests/components/ct-react-vite/src/components/Counter.tsx:20]: Yeah agree, i can change it to `span` instead.

Svelte complained about having a click event on a `div` so i changed everything to `button`. This also worked out well for the added tests.

Comment (pavelfeldman) [reply tests/components/ct-react-vite/src/components/Counter.tsx:20]: I'm not entirely sure where this is coming from (this does not sound like the right a11y advice to me). Maybe add a role="button" on your fix to make a11y tree happy?

Comment (sand4rt) [reply tests/components/ct-react-vite/src/components/Counter.tsx:20]: The warning is not that clear but it is correct. I believe it's about having a click event on divs and divs can't be triggered with keyboard events like `enter`. Adding a `role="button"` does not solve this problem.

Is there something wrong with a `span` in a `button`?

By the way, the a11y warning can be seen here: https://github.com/microsoft/playwright/actions/runs/4817238735/jobs/8577703732?pr=22678#step:8:360

Comment (pavelfeldman) [reply tests/components/ct-react-vite/src/components/Counter.tsx:20]: > I believe it's about having a click event on divs and divs can't be triggered with keyboard events like enter. Adding a role="button" does not solve this problem.

Yes, but that's a fairly naive take on a11y. Can these warnings be disabled?

> Is there something wrong with a span in a button?

Either way is acceptable, div was a bit unusual. The meta-point is: you are modifying the code due to the audit warning generated by the third party, implying that you trust the warning. But that warning does not sound right to me (and I've spent some time dealing with a11y). Let alone that our code is for testing. So I'd just disable it.

**Pull request merged**